medmodels/treatment_effect_estimation/analysis_modules/estimate.py

```diff
@@ -0,0 +1,407 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Dict, Set, Tuple
+
+from medmodels.medrecord.medrecord import MedRecord
+from medmodels.medrecord.types import NodeIndex
+from medmodels.treatment_effect_estimation.analysis_modules.matching.matching import (
+    Matching,
+)
+from medmodels.treatment_effect_estimation.analysis_modules.matching.neighbors import (
+    NeighborsMatching,
+)
+from medmodels.treatment_effect_estimation.analysis_modules.matching.propensity import (
+    PropensityMatching,
+)
+
+if TYPE_CHECKING:
+    from medmodels.treatment_effect_estimation.treatment_effect import TreatmentEffect
+
+
+class Estimate:
+    _treatment_effect: TreatmentEffect
+
+    def __init__(self, treatment_effect: TreatmentEffect) -> None:
+        self._treatment_effect = treatment_effect
+
+    def _check_medrecord(self, medrecord: MedRecord) -> None:
+        """Checks if the required groups are present in the MedRecord.
+
+        Args:
+            medrecord (MedRecord): The MedRecord object containing the data.
+
+        Raises:
+            AssertionError: If the required groups are not present in the MedRecord.
+        """
+        assert self._treatment_effect._patients_group in medrecord.groups, (
+            f"Patient group {self._treatment_effect._patients_group} not found in the "
+            f"data. Available groups: {medrecord.groups}"
+        )
+        assert self._treatment_effect._treatments_group in medrecord.groups, (
+            "Treatment group not found in the data. "
+            f"Available groups: {medrecord.groups}"
+        )
+        assert self._treatment_effect._outcomes_group in medrecord.groups, (
+            "Outcome group not found in the data. "
+            f"Available groups: {medrecord.groups}"
+        )
+
+    def _sort_patients_in_contingency_table(
+        self, medrecord: MedRecord
+    ) -> Tuple[Set[NodeIndex], Set[NodeIndex], Set[NodeIndex], Set[NodeIndex]]:
+        """Sorts patients into the contingency table of treatment-outcome, treatment-
+            no outcome, control-outcome and control-no outcome. The treatment group and
+            control matching is determined based on the treatment effect configuration.
+
+        Args:
+            medrecord (MedRecord): The MedRecord object containing the data.
+
+        Returns:
+            Tuple[Set[NodeIndex], Set[NodeIndex], Set[NodeIndex], Set[NodeIndex]: The
+                patient ids of true and false subjects in the treatment and control groups, respectively.
```

Comment:
Line too long

Reply:
there are also some other lines which are too long (mostly by one character). In this file as well as in other files. All of these lines should be a maximum of 88 characters long